packages/nameguard-js/src/canonical.ts

```diff
@@ -0,0 +1,59 @@
+import { ens_normalize } from "@adraffy/ens-normalize";
+import { splitGraphemes } from "./graphemes";
+import { graphemeConfusableAnalysis } from "./confusables";
+
+/**
+ * Returns the canonical form of the input label.
+ *
+ * A canonical label is the label where all confusable graphemes are replaced with their canonical form.
+ *
+ * A canonical grapheme is a grapheme that is considered the "base" version of a grapheme, for example, a Latin letter without diacritics.
+ * Some graphemes have no canonical form because of language-specific writing scripts.
```

Comment:
Some graphemes do not have a canonical because what people consider the "default" might depend on the language they use. Maybe @djstrong can better explain why there are graphemes without canonicals.

Reply:
I added this comment: Some graphemes have no canonical form when this form cannot be unambiguously determined.